File: Javascript/1401-1450/1436. Destination City.js
// https://leetcode.com/problems/destination-city/

var destCity = function(paths) {
    if(paths.length == 1) return paths[0][1];

    let destination = paths[0][1];
    const storage = new Map();
    
    for(const path of paths) // add all paths and destination to map
        storage.set(path[0],path[1]);

    while(storage.has(destination))
        destination = storage.get(destination);

    return destination;
};

File: Python/1401-1450/1436. Destination City.py
# https://leetcode.com/problems/destination-city/

class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        if len(paths) == 1:
            return paths[0][1]

        destination = paths[0][1]
        storage = {}
        
        for path in paths: # add all paths and destination to map
            storage[path[0]] = path[1]

        while(destination in storage):
            destination = storage[destination]

        return destination
